lib/puma/cluster.rb

```diff
@@ -127,6 +127,12 @@ def hup
         Process.kill "HUP", @pid
       rescue Errno::ESRCH
       end
+
+      def promote
+        Process.kill "URG", @pid
+        Process.detach @pid
```

Comment:
Process.detach should never be used. It's implementation is extremely bad as it spawns a Thread just to perform a wait in a loop.